src/hypothesis/internal/reflection.py

```diff
@@ -99,9 +99,9 @@ def required_args(target, args=(), kwargs=()):
         return None
     # For classes, self is present in the argspec but not really required
     posargs = spec.args[1:] if inspect.isclass(target) else spec.args
```

Comment:
While staring at this to make sure I understood what was going on I spotted another case this doesn't handle correctly. However it works with `builds()` introspection and I haven't quite figured out why!

The case is that of bound methods: self will appear in their argspec but isn't required.

I'm also finding this method's docstring somewhat less than informative. What's it actually supposed to return? What are args and kwargs supposed to be and how do they affect that?

Is the goal of this method actually that it returns required args that have not been supplied? Particularly given that there seems to have been some confusion about what these arguments were meant to be that lead to this bug, I'd like to see the docstring improved.